File: app/src/main/java/at/shockbytes/corey/workout/WorkoutManager.java
package at.shockbytes.corey.workout;

import java.util.List;

import at.shockbytes.corey.common.core.workout.model.Exercise;
import at.shockbytes.corey.common.core.workout.model.Workout;
import at.shockbytes.corey.storage.live.LiveWorkoutUpdateListener;
import io.reactivex.Observable;

/**
 * @author Martin Macheiner
 *         Date: 21.02.2017.
 */

public interface WorkoutManager {

    void poke();

    void addWorkout(Workout w);

    void deleteWorkout(Workout w);

    void updateWorkout(Workout w);

    Observable<List<Workout>> getWorkouts();

    Observable<List<Exercise>> getExercises();

    void updatePhoneWorkoutInformation(int workouts, int workoutTime);

    void registerLiveForWorkoutUpdates(LiveWorkoutUpdateListener listener);

    void unregisterLiveForWorkoutUpdates(LiveWorkoutUpdateListener listener);

}

File: app/src/main/java/at/shockbytes/corey/workout/WorkoutManager.kt
package at.shockbytes.corey.workout

import at.shockbytes.corey.common.core.workout.model.Exercise
import at.shockbytes.corey.common.core.workout.model.Workout
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 21.02.2017.
 */

interface WorkoutManager {

    val workouts: Observable<List<Workout>>

    val exercises: Observable<List<Exercise>>

    fun poke()

    fun storeWorkout(workout: Workout)

    fun deleteWorkout(workout: Workout)

    fun updateWorkout(workout: Workout)

    fun updatePhoneWorkoutInformation(workouts: Int, workoutTime: Int)

    fun updateWearWorkoutInformation(avgPulse: Int, workoutsWithPulse: Int, workoutTime: Int)

    fun registerLiveWorkoutUpdates(listener: LiveWorkoutUpdateListener)

    fun unregisterLiveWorkoutUpdates()

}
